### ISSUE TITLE
[telemetry] Instrument PhotonWorkerPool with spans and metrics

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#48]
## Goal

Instrument `PhotonWorkerPool` in `ph-oton-io` with ph-telemetry spans and metrics.

**Best effort-to-value ratio in the repository.** All four methods already measure the duration with a `StopWatch`, already carry a caller-supplied `sActionName`, and already catch and swallow failures — the instrumentation is close to mechanical, and it lights up *all* asynchronous work in a ph-oton application at once.

## Where

`ph-oton-io/src/main/java/com/helger/photon/io/PhotonWorkerPool.java`

```java
public CompletableFuture <Void> run (@NonNull String sActionName, @NonNull Runnable aRunnable)
public CompletableFuture <Void> runThrowing (@NonNull String sActionName, @NonNull IThrowingRunnable <? extends Exception> aRunnable)
public <T> CompletableFuture <T> supply (@NonNull String sActionName, @NonNull Supplier <T> aSupplier)
public <T> CompletableFuture <T> supplyThrowing (@NonNull String sActionName, @NonNull IThrowingSupplier <T, ? extends Exception> aSupplier)
```

All four share the same body shape:

```java
return CompletableFuture.runAsync ( () -> {
  final StopWatch aSW = StopWatch.createdStarted ();
  CONDLOG.info ( () -> "Starting '" + sActionName + "'");
  try { aRunnable.run (); }
  catch (final RuntimeException ex) { CONDLOG.error (...); }
  finally { aSW.stop (); CONDLOG.info (...); }
}, m_aES).exceptionally (ex -> { LOGGER.error (...); return null; });
```

## Why this is a good span

The whole lambda body runs on **one** pool thread, so `Telemetry.withSpanVoid (...)` / `withSpan (...)` inside the lambda starts and ends the span on the same thread. That satisfies the thread-binding constraint documented at the end of `LongRunningJobTelemetry.onJobEnd` — no manual `startSpan` bookkeeping is needed here.

Note the two distinct failure paths: the `catch` **inside** the lambda (which swallows the exception and lets the future complete normally) and the `.exceptionally (...)` handler **outside** it. Only the inner one is inside the span. The outer handler catches failures the inner `catch` did not — record those as a counter increment, not on the span, because the span is already closed by then.

## What to emit

New `com.helger.photon.io.CIOTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_WORKER_EXECUTE` | `photon.worker.execute` |
| `METRIC_WORKER_STARTED` | `photon.worker.started` |
| `METRIC_WORKER_ENDED` | `photon.worker.ended` |
| `METRIC_WORKER_RUNNING` | `photon.worker.running` |
| `METRIC_WORKER_DURATION` | `photon.worker.duration` |
| `ATTR_WORKER_ACTION` | `photon.worker.action` |
| `ATTR_WORKER_SUCCESS` | `photon.worker.success` |
| `UNIT_TASK` | `{task}` |
| `UNIT_MILLIS` | `ms` |

Plus a `PhotonWorkerPoolMetrics` instrument holder and a package-private emitter, mirroring `ph-oton-mgrs/.../longrun/` — which is the closest existing analogue, since this is the same "start / run / end with duration" shape.

- **Span** `photon.worker.execute`, kind `INTERNAL`, attribute `action`.
- **Counter** `photon.worker.started` / `photon.worker.ended` (the latter with a `success` attribute).
- **Up-down counter** `photon.worker.running` — `+1` at the start of the lambda, `-1` in the `finally`. Both sides must use **identical** attributes so it nets out to zero, exactly as `LongRunningJobMetrics.JOBS_RUNNING` documents.
- **Histogram** `photon.worker.duration` in ms, from the existing `aSW`, attributes `action` + `success`.

## Cardinality — the one real risk here

`sActionName` is caller-supplied free text. Most existing callers pass a constant, but nothing enforces that, and a caller interpolating an ID (`"import of " + sDocID`) would blow up the metric cardinality.

Handle it explicitly rather than hoping:
- put `sActionName` on the **span** unconditionally, and
- for the **metrics**, either document loudly in the Javadoc of `run`/`supply` that `sActionName` must be a bounded constant, or clamp it (e.g. drop the attribute when it exceeds a length limit).

Pick one during implementation and write down the choice in `CIOTelemetry`'s Javadoc. Do not leave it unaddressed.

## Refactoring note

The four methods are near-identical. Extracting the shared span+metric wrapper into one private helper before instrumenting will avoid four copies of the same code. Keep the four public signatures unchanged.

## POM

`ph-oton-io/pom.xml` needs a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

`ph-oton-io` is the lowest layer that would get `ph-telemetry` — it currently depends only on `ph-collection`, `ph-xml` and `ph-dao`. That is fine: `ph-telemetry` itself depends only on `ph-annotations` + `ph-base`, and degrades to no-ops without an SPI.

## Acceptance criteria

- [ ] All four methods instrumented, ideally through one shared private helper.
- [ ] Up-down counter uses identical attributes on the `+1` and `-1` sides.
- [ ] The `.exceptionally (...)` path increments the ended-counter with `success=false`.
- [ ] The `sActionName` cardinality decision is implemented and documented in `CIOTelemetry`.
- [ ] Unit test with recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)` and reset to `null` in `@After`, covering a successful and a throwing task.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.6.0**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-io/.../io/CIOTelemetry.java` | new, public - span / metric / attribute names |
| `ph-oton-io/.../io/PhotonWorkerPoolMetrics.java` | new, public - 2 counters, 1 up-down counter, 1 histogram |
| `ph-oton-io/.../io/PhotonWorkerPoolTelemetry.java` | new, package-private emitter |
| `ph-oton-io/.../io/PhotonWorkerPool.java` | modified - four near-identical bodies collapsed into one helper |
| `ph-oton-io/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |
| `ph-oton-io/src/test/.../PhotonWorkerPoolTelemetryTest.java` | new, 3 tests |

### The refactoring

As suggested, the four public methods now share one private helper and keep their signatures:

```java
@Nullable
private static <T> T _executeInstrumented (String sActionName,
                                           String sTaskType,
                                           Object aTask,
                                           AtomicBoolean aEndEmitted,
                                           IThrowingSupplier <T, Exception> aBody)
```

`run` and `runThrowing` pass `() -> { aRunnable.run (); return null; }`, `supply` and `supplyThrowing` pass `aSupplier::get`. The four `.exceptionally (...)` handlers likewise collapse into one `_onUnexpectedException (...)` factory. The `catch (RuntimeException)` of the two non-throwing variants became `catch (Exception)`, which is behaviourally identical there because a `Runnable` / `Supplier` cannot throw a checked exception. Every log message is byte-for-byte what it was.

### Cardinality - the decision, and why it changed

The issue offered two options for `sActionName`: document loudly, or clamp. **Neither was chosen: the action name is a span attribute only and is never a metric attribute.**

The reason is that the risk is not hypothetical - it is already realised in production code. phoss-smp does exactly this:

```java
PhotonWorkerPool.getInstance ()
                .run ("Add '" + aPID.getURIEncoded () + "' to the " + SMPWebAppConfiguration.getDirectoryName (),
                      () -> aPDClient.addServiceGroupToIndex (aPID));
```

One time series per Peppol participant ID. Documenting a constraint that the ecosystem's own flagship application already violates would not have helped, and a length clamp does not detect unboundedness (`"import 4711"` is short). So `photon.worker.started` / `ended` / `running` / `duration` carry at most a `success` attribute, and the per-action breakdown is available from the span - which is where an unbounded dimension is affordable. The reasoning is written down in the Javadoc of `CIOTelemetry.ATTR_WORKER_ACTION`, complete with the phoss-smp example.

If a per-action *metric* breakdown is wanted later, the right shape is an opt-in overload that takes a separate bounded metric name - not a reinterpretation of the existing free-text parameter.

### Exactly one end-of-task measurement

Both the inner `finally` and the outer `.exceptionally (...)` can fire for the same task (an `Error` escapes the body: the finally runs, then the future completes exceptionally). They now share an `AtomicBoolean` guard, so `photon.worker.ended` is emitted exactly once per task, and `photon.worker.running` is always balanced. `PhotonWorkerPoolTelemetryTest.testThrowingSupplier` asserts that there is exactly one ended-measurement.

The remaining asymmetry is deliberate and documented on `METRIC_WORKER_ENDED`: a task that never reached a worker thread at all - executor already shut down - produces an `ended` without a `started`. That difference is exactly the number of silently dropped tasks and is the interesting signal.

### Verification

`PhotonWorkerPoolTelemetryTest`: 3/3 green - a successful task, a throwing task (future completes normally, span marked `error`, duration recorded with `success=false`), and the exactly-once check. It asserts that the up-down counter nets out to zero and that the action name never appears in a metric attribute map. Full reactor `mvn install` over all 23 modules: BUILD SUCCESS. `README.md` news bullet added under `v10.6.0 - work in progress`.
